src/core/etl/tests/Flow/ETL/Tests/Integration/Filesystem/FilesystemStreams/Partitioned/OverwriteModeTest.php

```diff
@@ -85,6 +86,29 @@ public function test_open_stream_for_non_existing_partition() : void
         self::assertSame('new content', \file_get_contents($files[0]->path->path()));
     }
 
+    public function test_open_stream_for_non_existing_partition_with_custom_schema() : void
+    {
+        $this->setupFiles([__FUNCTION__ => []]);
+
+        $fs = new FakeNativeLocalFilesystem();
```

Comment:
could this be replaced with `src/lib/filesystem/src/Flow/Filesystem/Local/MemoryFilesystem.php` maybe? I'm not sure how I feel about Fake Filesystem in ETL package

Reply:
Memory doesn't support `mv()` method, so it will unfortunately not help :(